astropy/time/core.py

```diff
@@ -674,9 +674,10 @@ def format(self, format):
         self._format = format
 
     def __repr__(self):
-        return ("<{} object: scale='{}' format='{}' value={}>"
+        with np.printoptions(threshold=10, floatmode='fixed'):
```

Comment:
I would propose to set the threshold to 50, remove that `floatmode` setting, and call this done.  If somebody wants to have `floatmode='fixed'` formatting they can use a `np.printoptions` context manager around the repr statement.